src/generator.rs

```diff
@@ -44,26 +46,36 @@ pub fn run_generator(root: &Path, devices: &[Device], output_directory: &PathBuf
         return Ok(());
     }
 
-    let mut devices_made = false;
-    for dev in devices {
-        devices_made |= handle_device(output_directory, dev)?;
-    }
-    if devices_made {
-        /* We created some devices, let's make sure the module is loaded */
-        let modules_load_path = root.join("run/modules-load.d/zram-generator.conf");
-        make_parent(&modules_load_path)?;
-        fs::write(&modules_load_path, "zram\n").with_context(|| {
-            format!(
-                "Failed to write configuration for loading a module at {}",
-                modules_load_path.display()
-            )
-        })?;
+    let devices_made: Vec<_> = Result::from_iter(
+        devices
+            .iter()
+            .map(|dev| handle_device(output_directory, dev)),
+    )?;
+    if !devices_made.is_empty() {
+        /* We created some devices, let's make sure the module is loaded and they exist */
+        if let Err(e) = Command::new("modprobe").arg("zram").status() {
+            return Err(anyhow!("modprobe call failed: {}", e));
+        }
+
+        let max_device = devices_made.into_iter().fold(0, cmp::max);
+        if !Path::new("/dev")
+            .join(format!("zram{}", max_device))
+            .exists()
+        {
+            while fs::read_to_string("/sys/class/zram-control/hot_add")
+                .context("Adding zram device")?
+                .trim_end()
+                .parse::<u64>()
+                .context("Fresh zram device number")?
```

Comment:
I don't want this to happen in the "generation" phase. Loading of modules is a potentially lengthy operation (because the kernel might need to pull in dependencies first, read stuff off the filesystem, etc). Generators are supposed to run quickly and without blocking.

Instead, this should be moved to the setup phase.

Reply:
As noted in https://github.com/systemd/zram-generator/issues/14#issuecomment-640725858, the setup phase will simply *never run* if we don't do this, just like it doesn't now for zram devs higher than the default num_devices (usually 1).